File: backend/domain/table/lieferung.go
package table

import (
	"time"

	z "github.com/Oudwins/zog"
)

type Lieferung struct {
	ID          string
	UserID      int
	TischID     int
	Positionen  []Position
	Kommentar   string
	GeliefertAm time.Time
}

var lieferungSchema = z.Struct(z.Shape{
	"ID":          z.String().UUID().Required(),
	"UserID":      z.Int().GTE(1).Required(),
	"TischID":     z.Int().GTE(1).Required(),
	"Positionen":  z.Slice(positionSchema).Min(1).Required(),
	"Kommentar":   z.String().Max(100),
	"GeliefertAm": z.Time().Required(),
})

File: frontend/src/service/table/Lieferung.ts
import { z } from 'zod'

import { PositionRefSchema } from './Bestellung'

export const LieferungSchema = z.object({
  id: z.uuid(),
  userId: z.number().int().min(1),
  tischId: z.number().int().min(1),
  positionen: PositionRefSchema.array().min(1),
  kommentar: z.string().max(100),
  geliefertAm: z.string().refine((date) => !isNaN(Date.parse(date)), {
    message: 'Invalid date format',
  }),
})
export type Lieferung = z.infer<typeof LieferungSchema>

export const ProdukteLiefernSchema = z.object({
  tischId: z.number().int().min(1),
  positionen: PositionRefSchema.array().min(1),
  kommentar: z.string().max(100),
})
export type ProdukteLiefern = z.infer<typeof ProdukteLiefernSchema>
